API.md

```diff
@@ -0,0 +1,269 @@
+# API
+
+* [`create`](#create)
+* [`start`](#start)
+* [`stop`](#stop)
+* [`dial`](#dial)
+* [`dialProtocol`](#dialProtocol)
+* [`hangUp`](#hangUp)
+* [`handle`](#handle)
+* [`unhandle`](#unhandle)
+* [`peerRouting.findPeer`](#peerRouting.findPeer)
+* [`contentRouting.findProviders`](#contentRouting.findProviders)
+* [`contentRouting.provide`](#contentRouting.provide)
+* [`dht.put`](#dht.put)
```

Comment:
Suggestion: Avoid exposing the DHT methods as they are just methods from one impl of Peer Routing and Content Routing and instead stick with a clean Peer Routing & Content Routing interfaces.

Reply:
Shouldn't we make that clear instead?

As there are projects, including `IPFS` that use the `dht` API directly [interface-js-ipfs-core/SPEC/DHT.md](https://github.com/ipfs/interface-js-ipfs-core/blob/master/SPEC/DHT.md), I think we should make your point clear in the docs, but also have the `dht` available

Reply:
A way to declare that the API is internal is by putting a `_` before the name (e.g. `_dht`), this way, folks are less tempted to use it.